pkg/v1/remote/write.go

```diff
@@ -296,4 +291,22 @@ func (w *writer) commitImage() error {
 	return nil
 }
 
+func scopesForUploadingImage(ref name.Reference, layers []v1.Layer) []string {
+	// use a map as set to remove duplicates scope strings
+	scopeSet := map[string]struct{}{ref.Scope(transport.PushScope): {}}
```

Comment:
So this is kind of annoying (and not your fault), but I believe we want the `PushScope` to be the first scope due to how some registries are implemented. I believe a few registries just look at the first scope to determine access (this is a bug), and since PushScope is the broadest (it has both `push,pull`), we need it to be first in order to have write access.

So tl;dr, we want the push scope to be first, but we don't care about the order of any of the other scopes (just read access).